silo-core/contracts/incentives/interfaces/IDistributionManager.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.28;
+
+import {DistributionTypes} from "../lib/DistributionTypes.sol";
+
+interface IDistributionManager {
+    struct IncentivesProgram {
+        uint256 index;
+        address rewardToken; // can't be updated after creation
+        uint104 emissionPerSecond; // configured by owner
+        uint40 lastUpdateTimestamp;
+        uint40 distributionEnd; // configured by owner
+        mapping(address => uint256) users;
+    }
+
+    struct IncentiveProgramDetails {
+        uint256 index;
+        address rewardToken;
+        uint104 emissionPerSecond;
+        uint40 lastUpdateTimestamp;
+        uint40 distributionEnd;
+    }
+
+    struct AccruedRewards {
```

Comment:
let's add user address

Reply:
in general, maybe we should address problem we have with current incentive controllel: we should be able to read events how much user get for each asset (if possible)

Reply:
It is already implemented. Take a look at the `RewardsAccrued` event:
`event RewardsAccrued(address indexed user, address indexed rewardToken, uint256 amount);`

Reply:
let me be more precise: how you will tell that for particular Silo user got X rewards? you can only tell the sum of rewards based on event, not amount per market.

Reply:
It is not an issue, as we have one incentive controller per share token.